java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java

```diff
@@ -770,29 +770,36 @@ void readRanges(FSDataInputStream file, boolean allocateDirect, double extraByte
     }
 
     static ChunkReader create(BufferChunk from, BufferChunk to) {
-      long f = Integer.MAX_VALUE;
-      long e = Integer.MIN_VALUE;
+      long start = Long.MAX_VALUE;
+      long end = Long.MIN_VALUE;
 
-      long cf = Integer.MAX_VALUE;
-      long ef = Integer.MIN_VALUE;
-      int reqBytes = 0;
+      long currentStart = Long.MAX_VALUE;
+      long currentEnd = Long.MIN_VALUE;
+      long reqBytes = 0L;
 
       BufferChunk current = from;
       while (current != to.next) {
-        f = Math.min(f, current.getOffset());
-        e = Math.max(e, current.getEnd());
-        if (ef == Integer.MIN_VALUE || current.getOffset() <= ef) {
-          cf = Math.min(cf, current.getOffset());
-          ef = Math.max(ef, current.getEnd());
+        start = Math.min(start, current.getOffset());
+        end = Math.max(end, current.getEnd());
+        if (currentEnd == Long.MIN_VALUE || current.getOffset() <= currentEnd) {
+          currentStart = Math.min(currentStart, current.getOffset());
+          currentEnd = Math.max(currentEnd, current.getEnd());
         } else {
-          reqBytes += ef - cf;
-          cf = current.getOffset();
-          ef = current.getEnd();
+          reqBytes += currentEnd - currentStart;
+          currentStart = current.getOffset();
+          currentEnd = current.getEnd();
         }
         current = (BufferChunk) current.next;
       }
-      reqBytes += ef - cf;
-      return new ChunkReader(from, to, (int) (e - f), reqBytes);
+      reqBytes += currentEnd - currentStart;
+      if (reqBytes >= Integer.MAX_VALUE) {
```

Comment:
https://github.com/apache/orc/blob/22a06b4e7575f539be970ec9658f5b597ef7c83d/java/core/src/java/org/apache/orc/impl/StringHashTableDictionary.java#L61-L65
I think it might be more accurate to extract and reuse this `MAX_ARRAY_SIZE` constant, the java maximum array length is not actually `Integer.MAX_VALUE`.

Reply:
Should I define a `MAX_ARRAY_SIZE` constant in `RecordReaderUtils` ? I noticed it is marked as `private` in `StringHashTableDictionary`.

Reply:
I tend to extract it into a utility class (e.g. IOUtils) and reuse the same constant.

Reply:
I agree "the same concept should be kept in a single source".
         
Should we include the changes to remove the `MAX_ARRAY_SIZE` definition from `StringHashTableDictionary.java` in this PR?

Reply:
Considering that this PR has been closed, is it still necessary to continue the discussion on the issues mentioned above?